app/controllers/conversations.js

```diff
@@ -201,67 +226,154 @@ class ConversationsController extends BaseController {
 
     let isOwnerChange = false;
     if (
+      conversation.type !== "u" &&
       requestData.participants &&
       Object.keys(requestData.participants) != 0
     ) {
       const participantsToUpdate = requestData.participants;
       delete requestData.participants;
 
+      let currentUserParams;
       const { add: addUsers, remove: removeUsers } = participantsToUpdate;
       const countParticipants = await ConversationParticipant.count({
         conversation_id: conversationId,
       });
-      if (addUsers) {
+
+      let existingParticipantsIds = (
+        await ConversationParticipant.findAll(
+          { conversation_id: conversationId },
+          ["user_id"]
+        )
+      ).map((p) => p.user_id.toString());
+
+      let newParticipantsIds = addUsers
+        ? addUsers
+            .map((uId) => uId.toString())
+            .filter((uId) => !existingParticipantsIds.includes(uId))
+        : [];
+
+      let removeParticipantsIds = removeUsers
+        ? removeUsers
+            .map((uId) => uId.toString())
+            .filter((uId) => existingParticipantsIds.includes(uId))
+        : [];
+
+      let removeParticipantsInfo = [];
+      let newParticipantsInfo =
+        newParticipantsIds.length || removeParticipantsIds.length
+          ? (
+              await User.findAll(
+                {
+                  _id: {
+                    $in: [
+                      currentUserId,
+                      ...newParticipantsIds,
+                      ...removeParticipantsIds,
+                    ],
+                  },
+                },
+                [
+                  "login",
+                  "first_name",
```

Comment:
Would be great to move this newly added code for `newParticipantsIds ` calculation to somewhere else place. It becomes messy... 

I think it's a good time to introduce services. 
Let's create a `app/services/conversation.js` and move this code there